pkg/resources/management.cattle.io/v3/node/validator.go

```diff
@@ -0,0 +1,63 @@
+package node
+
+import (
+	"fmt"
+
+	"github.com/rancher/webhook/pkg/admission"
+	objectsv3 "github.com/rancher/webhook/pkg/generated/objects/management.cattle.io/v3"
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var nodeGVR = schema.GroupVersionResource{
+	Group:    "management.cattle.io",
+	Version:  "v3",
+	Resource: "nodes",
+}
+
+const localNs = "local"
+
+type Validator struct {
+	admitter admitter
+}
+
+type admitter struct{}
+
+// Admit is the entrypoint for the validator. It lets a request through or prevent user from executing it.
+func (a admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResponse, error) {
+	oldNode, _, err := objectsv3.NodeOldAndNewFromRequest(&request.AdmissionRequest)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get old and new node from request: %w\n", err)
+	}
+	if request.Operation == admissionv1.Delete && oldNode.Namespace == localNs {
```

Comment:
What if a user rotates all of the nodes in their local cluster? I don't think we should be leaving all of the nodes that have ever existed around. If they rotate nodes say once a month, after a year you could have 96 left over node objects. Rotate enough nodes, and you could have resource exhaustion in the local cluster. I would recommend updating the approach to maybe account for accidental deletion, but allow intentional deletion. i.e. if the corev1.Node object really doesn't exist anymore, then perhaps annotate the v3.Node object and gate the deletion validator behind the presence of the annotation for the local cluster.